src/test/javascript/spec/component/exam/manage/exam-management.service.spec.ts

```diff
@@ -0,0 +1,405 @@
+
+        // THEN
+        httpMock.expectOne({
+            method: 'POST',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/student-exams/start-exercises`,
+        });
+    });
+
+    it('should evaluate quiz exercises', () => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockEvaluatedExercises = 1;
+
+        // WHEN
+        service.evaluateQuizExercises(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.equal(mockEvaluatedExercises));
+
+        // THEN
+        httpMock.expectOne({
+            method: 'POST',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/student-exams/evaluate-quiz-exercises`,
+        });
+    });
+
+    it('should assess unsubmitted exam modelling and text participations', () => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockUnsubmittedExercises = 1;
+
+        // WHEN
+        service.assessUnsubmittedExamModelingAndTextParticipations(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.equal(mockUnsubmittedExercises));
+
+        // THEN
+        httpMock.expectOne({
+            method: 'POST',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/student-exams/assess-unsubmitted-and-empty-student-exams`,
+        });
+    });
+
+    it('should unlock all repositories', () => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockRepoCount = 1;
+
+        // WHEN
+        service.unlockAllRepositories(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.equal(mockRepoCount));
+
+        // THEN
+        httpMock.expectOne({
+            method: 'POST',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/student-exams/unlock-all-repositories`,
+        });
+    });
+
+    it('should lock all repositories', () => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockRepoCount = 1;
+
+        // WHEN
+        service.lockAllRepositories(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.equal(mockRepoCount));
+
+        // THEN
+        httpMock.expectOne({
+            method: 'POST',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/student-exams/lock-all-repositories`,
+        });
+    });
+
+    it('should update order', () => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockExerciseGroups: ExerciseGroup[] = [{ exam: mockExam, id: 1 }];
+
+        // WHEN
+        service.updateOrder(course.id!, mockExam.id!, mockExerciseGroups).subscribe((res) => expect(res.body).to.equal(mockExerciseGroups));
+
+        // THEN
+        httpMock.expectOne({
+            method: 'PUT',
+            url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/exerciseGroupsOrder`,
+        });
+    });
+});
+
+function mockConvertDateFromClient(exam: Exam): Exam {
```

Comment:
Why did you mock this? It would be useful to test the real function here

Reply:
Is it possible to test the real function (since it's private)?

Reply:
Afaik we can't test private functions, but would have to make it publicly accessible.

Reply:
Its not private in other services as far as I can see. Just remove the private and use the real function here imo